R/ml_estimation.R

```diff
@@ -141,510 +161,211 @@ ml_estimation.default <- function(x,
-      estimates_loc_sc[[1]] + stats::qnorm((1 + conf_level) / 2) * se_loc_sc[[1]])
-
-    w <- exp(
-      (stats::qnorm((1 + conf_level) / 2) * se_loc_sc[[2]]) / estimates_loc_sc[[2]])
-    conf_sig <- c(estimates_loc_sc[[2]] /  w, estimates_loc_sc[[2]] * w)
-
-    conf_gamma <- c(
-      estimates_loc_sc[[3]] + stats::qnorm((1 - conf_level) / 2) * se_loc_sc[[3]],
-      estimates_loc_sc[[3]] + stats::qnorm((1 + conf_level) / 2) * se_loc_sc[[3]])
-
-    conf_ints_loc_sc <- matrix(c(conf_mu, conf_sig, conf_gamma), byrow = TRUE, ncol = 2)
-    colnames(conf_ints_loc_sc) <- c(paste(((1 - conf_level) / 2) * 100,"%"),
-                                    paste(((1 + conf_level) / 2) * 100, "%"))
-    rownames(conf_ints_loc_sc) <- names(estimates_loc_sc)
-
-    if (distribution == "weibull3") {
-      # Alternative Parameterization, only for Weibull:
-      estimates <- c(exp(estimates_loc_sc[[1]]), 1 / estimates_loc_sc[[2]],
-                     estimates_loc_sc[[3]])
-      names(estimates) <- c("eta", "beta", "gamma")
-
-      conf_ints <- matrix(c(exp(conf_mu), rev(1 / conf_sig), conf_gamma), byrow = TRUE,
-                          ncol = 2)
-      colnames(conf_ints) <- colnames(conf_ints_loc_sc)
-      rownames(conf_ints) <- names(estimates)
-
-      ml_output <- list(
-        coefficients = estimates_loc_sc,
-        confint = conf_ints_loc_sc,
-        varcov = vcov_loc_sc,
-        shape_scale_coefficients = estimates,
-        shape_scale_confint = conf_ints,
-        logL = ml$value,
-        aic = -2 * (ml$value) + 2 * length(estimates_loc_sc),
-        bic = (-2 * (ml$value) + log(length(x)) * length(estimates_loc_sc))
-      )
-
-    } else {
-      ml_output <- list(
-        coefficients = estimates_loc_sc,
-        confint = conf_ints_loc_sc,
-        varcov = vcov_loc_sc,
-        logL = ml$value,
-        aic = -2 * (ml$value) + 2 * length(estimates_loc_sc),
-        bic = (-2 * (ml$value) + log(length(x)) * length(estimates_loc_sc))
-      )
-    }
-  }
-  # Location-Scale Models:
-  if (distribution %in% c("sev", "normal", "logistic")) {
-    # Location-Scale:
-    ## Problem:  With functions SPREDA::Lifedata.MLE it is not possible to estimate
-    ##           the parameters of a location-scale distribution and with survival::survreg it is
-    ##           not possible to estimate var-cov-matrix of location-scale parameters.
-    ## Solution: A preliminary parameter estimation is done with MLE (using survreg)
-    ##           for location-scale distributions.
-    ##           These estimates will be used as initial values to maximize the log-likelihood of
-    ##           location-scale distributions.
-
-    ## rename distributions:
-    if (distribution == "sev") {
-      distr = "extreme"
-    } else if (distribution == "normal") {
-      distr = "gaussian"
-    } else {
-      distr = distribution
+    ## Initial value for threshold parameter:
+    t0 <- 0
+    if (length(start_dist_params) == 3L) {
```

Comment:
Why do we need that check? I assume `check_dist_params()`, which was called earlier ensures that `start_dist_params` has length 3